src/ServiceControl/CustomChecks/ReportCustomCheckResultHandler.cs

```diff
@@ -37,13 +38,19 @@ public void Handle(ReportCustomCheckResult message)
                 throw new Exception("Received an custom check message without proper initialization of the HostId in the schema");
             }
 
+            HandleAsync(message).GetAwaiter().GetResult();
+        }
+
+        private async Task HandleAsync(ReportCustomCheckResult message)
+        {
```

Comment:
Should the validations also be moved into the async method?

Reply:
I can do it while merging if that is ok